app/javascript/mastodon/features/compose/components/language_dropdown.jsx

```diff
@@ -206,52 +138,89 @@ class LanguageDropdownMenu extends PureComponent {
 
       break;
     }
-  };
+  }, [listNodeRef, onChange, onClose, searchValue]);
 
-  handleClear = () => {
-    this.setState({ searchValue: '' });
-  };
+  const handleClear = useCallback(() => {
+    setSearchValue('');
+  }, [setSearchValue]);
 
-  renderItem = lang => {
-    const { value } = this.props;
+  const isSearching = searchValue !== '';
 
-    return (
-      <div key={lang[0]} role='option' tabIndex={0} data-index={lang[0]} className={classNames('language-dropdown__dropdown__results__item', { active: lang[0] === value })} aria-selected={lang[0] === value} onClick={this.handleClick} onKeyDown={this.handleKeyDown}>
-        <span className='language-dropdown__dropdown__results__item__native-name' lang={lang[0]}>{lang[2]}</span> <span className='language-dropdown__dropdown__results__item__common-name'>({lang[1]})</span>
-      </div>
-    );
-  };
-
-  render () {
-    const { intl } = this.props;
-    const { searchValue } = this.state;
-    const isSearching = searchValue !== '';
-    const results = this.search();
-
-    return (
-      <div ref={this.setRef}>
-        <div className='emoji-mart-search'>
-          <input type='search' value={searchValue} onChange={this.handleSearchChange} onKeyDown={this.handleSearchKeyDown} placeholder={intl.formatMessage(messages.search)} />
-          <button type='button' className='emoji-mart-search-icon' disabled={!isSearching} aria-label={intl.formatMessage(messages.clear)} onClick={this.handleClear}><Icon icon={!isSearching ? SearchIcon : CancelIcon} /></button>
-        </div>
-
-        <div className='language-dropdown__dropdown__results emoji-mart-scroll' role='listbox' ref={this.setListRef}>
-          {results.map(this.renderItem)}
-        </div>
+  useEffect(() => {
+    document.addEventListener('click', handleDocumentClick, { capture: true });
```

Comment:
You can define `handleDocumentClick` locally within the `useEffect` and avoid having to use `useCallback` for it.